### PR TITLE
⚗️ Add password authentication for redis ( :warning: OPS)

### PR DESCRIPTION
**Pull request by mrnicegyu11**
** Follow-Up PR for https://github.com/ITISFoundation/osparc-simcore/pull/5708**
## What do these changes do?
We would like to use the AWS managed, fully redis-compatible "MemoryDB" in AWS, since we have now introduced non-ephemeral data in redis that should strictly persist even across incidents (CC @GitHK for details). AWS provides MemoryDB (see https://docs.aws.amazon.com/memorydb/latest/devguide/what-is-memorydb-for-redis.html ) as an endpoint/URL, that redis-clients can connect to *with password authentication*. Up until now, redis was running inside the `simcore` docker stack, isolated from outside connections via docker networks, and simcore services would connect to it directly, unencrypted and without authentication. While there was some infrastructure in the code to allow password authentication, it was not working out-of-the-box.

This PR will change the communication of simcore services with redis to always use password authentication.

**Bonus**:
- Sets the timeout of the integration test for the webserver-exporter to 60s (before: 10s), since it would on my machine always need at least 30s timeout to manage to export the project, and since we are not using the exporter anywhere. My hope is that this reduces flakyness of this test. I can revert this bonus-change if needed.

## Related issue/s
- https://github.com/ITISFoundation/osparc-ops-environments/issues/241
- Follow-Up PR for https://github.com/ITISFoundation/osparc-simcore/pull/5708

## Related PRs
- Redis-commander also uses password: https://github.com/ITISFoundation/osparc-ops-environments/pull/719
- Env-Vars: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/720
- Env-Vars: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/721
- Env-Vars: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/722

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))
- ENV-VARS master (merge asap): https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/720
- ENV-VARS staging: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/721
- ENV-VARS production: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/722

### CODE REVIEW

**Review by YuryHrytsuk — APPROVED** [review]
Thanks!

File: services/web/server/src/simcore_service_webserver/application_settings_utils.py [hunk 1]
```diff
@@ -75,6 +75,9 @@ def convert_to_app_config(app_settings: ApplicationSettings) -> dict[str, Any]:
                 "enabled": app_settings.WEBSERVER_REDIS is not None,
                 "host": getattr(app_settings.WEBSERVER_REDIS, "REDIS_HOST", None),
                 "port": getattr(app_settings.WEBSERVER_REDIS, "REDIS_PORT", None),
+                "password": getattr(
+                    app_settings.WEBSERVER_REDIS, "REDIS_PASSWORD", None
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/application_settings_utils.py:79]: i guess you might also need here to `get_secret_valu3` ... although probably this part of the code is not anymore used. I think it was deprecated.

File: services/web/server/src/simcore_service_webserver/application_settings_utils.py [hunk 2]
```diff
@@ -242,6 +245,7 @@ def _set_if_disabled(field_name, section):
             _set_if_disabled("WEBSERVER_REDIS", section2)
             envs["REDIS_HOST"] = section2.get("host")
             envs["REDIS_PORT"] = section2.get("port")
+            envs["REDIS_PASSWORD"] = section2.get("password")
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/application_settings_utils.py:248]: same here. Double check whether you need to call `get_secret_value`

File: services/web/server/tests/unit/with_dbs/conftest.py [hunk 3]
```diff
@@ -87,12 +87,14 @@ def disable_swagger_doc_generation(
...
     postgres_cfg = default_app_cfg["db"]["postgres"]
-
+    redis_cfg = default_app_cfg["resource_manager"]["redis"]
     # docker-compose reads these environs
     with pytest.MonkeyPatch().context() as patcher:
         patcher.setenv("TEST_POSTGRES_DB", postgres_cfg["database"])
         patcher.setenv("TEST_POSTGRES_USER", postgres_cfg["user"])
         patcher.setenv("TEST_POSTGRES_PASSWORD", postgres_cfg["password"])
+        # Redis
+        patcher.setenv("TEST_REDIS_PASSWORD", redis_cfg["password"])
```
Comment (pcrespov) [inline services/web/server/tests/unit/with_dbs/conftest.py:97]: remember my comment before about `get_secret_value`. My guess is that here you are injecting `*******` . 
pls double check.

File: services/web/server/tests/unit/with_dbs/docker-compose-devel.yml
Comment (pcrespov) [inline services/web/server/tests/unit/with_dbs/docker-compose-devel.yml:1]: My guess is that this docker-compose is also not used. We should do some cleanup if so

**Review by pcrespov — APPROVED** [review]
Thx. Left some suggestions . please consider them before merging.

File: services/web/server/tests/integration/01/test_exporter_requests_handlers.py [hunk 4]
```diff
@@ -247,7 +247,7 @@ def _get_header_params(header):
 
     assert url_export == URL(f"/{API_VTAG}/projects/{project_id}:xport")
     async with await client.post(
-        f"{url_export}", headers=headers, timeout=10
+        f"{url_export}", headers=headers, timeout=60
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/web/server/tests/integration/01/test_exporter_requests_handlers.py:250]: Also, if this exporter code is unused. it should be cleaned away @GitHK @pcrespov, this only uses time/resources for nothing, or is it still used somewhere?

File: services/docker-compose.yml [hunk 5]
```diff
@@ -1132,15 +1141,17 @@ services:
...
+        "${REDIS_PASSWORD}"
       ]
     networks:
       - default
       - autoscaling_subnet
     volumes:
       - redis-data:/data
     healthcheck:
-      test: [ "CMD", "redis-cli", "ping" ]
+      test: [ "CMD", "redis-cli", "-a", "${REDIS_PASSWORD}", "ping" ]
```
Comment (sanderegg) [inline services/docker-compose.yml:1154]: no long named option instead of -a ?

**Review by sanderegg — APPROVED** [review]
thanks, added a few comments, but looks good.

File: services/web/server/tests/integration/01/test_exporter_requests_handlers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply services/web/server/tests/integration/01/test_exporter_requests_handlers.py:250]: @sanderegg @pcrespov : I will revert my change from `10` to `60` here then.

File: services/docker-compose.yml (hunk 5 above)
Comment (mrnicegyu11) [reply services/docker-compose.yml:1154]: I double-checked an it seems that there is no second, long-name-style option subsituting "-a"

**Pull request merged**